src/service/debtor-service.ts

```
@@ -0,0 +1,303 @@
+          }
+        }
+
+        const transfer = await TransferService.createTransfer({
+          amount: amount.toObject(),
+          fromId: user.id,
+          description: `Fine for balance of ${dinero({ amount: b.amount.amount }).toFormat()} on ${date.toLocaleDateString()}.`,
+          toId: undefined,
+        });
+
+        emails.push({ user, email: new UserGotFined({
+          name: user.firstName,
+          fine: amount,
+          balance: DineroTransformer.Instance.from(b.amount.amount),
+          referenceDate: date,
+          totalFine: userFineGroup.fines.reduce((sum, f) => sum.add(f.amount), dinero({ amount :0 })).add(amount),
+        }) });
+
+        return Object.assign(new Fine(), {
+          fineHandoutEvent,
+          userFineGroup,
+          amount: calculateFine(b.amount),
+          previousFine,
+          transfer,
+        });
+      }));
+      return { fines: await manager.save(fines), fineHandoutEvent, emails };
+    });
+
+    emails1.forEach(({ user, email }) => Mailer.getInstance().send(user, email));
+
+    return {
+      id: fineHandoutEvent1.id,
+      createdAt: fineHandoutEvent1.createdAt.toISOString(),
+      updatedAt: fineHandoutEvent1.updatedAt.toISOString(),
+      referenceDate: fineHandoutEvent1.referenceDate.toISOString(),
+      createdBy: parseUserToBaseResponse(fineHandoutEvent1.createdBy, false),
+      fines: fines1.map((f) => this.asFineResponse(f)),
+    };
+  }
+
+  /**
+   * Delete a fine with its transfer, but keep the FineHandoutEvent (they can be empty)
+   * @param id
+   */
+  public static async deleteFine(id: number): Promise<void> {
+    const fine = await Fine.findOne({ where: { id }, relations: ['transfer', 'userFineGroup', 'userFineGroup.fines'] });
+    if (fine == null) return;
+
+    const { transfer, userFineGroup } = fine;
+
+    await Fine.remove(fine);
+    await Transfer.remove(transfer);
+    if (userFineGroup.fines.length === 1) await UserFineGroup.remove(userFineGroup);
```

Maybe throw an error message here if length !== 0?

---

Niet eens, want een length === 0 kan niet (want je haalt de `UserFineGroup` op via `Fine`, dus de `UserFineGroup` heeft sowieso één fine) en bij een lengte van 2 is er nog minimaal één fine in de groep die je niet gaat verwijderen.